scm-core/src/main/java/sonia/scm/repository/spi/BranchesCommand.java

```diff
@@ -41,13 +38,19 @@
 public interface BranchesCommand
 {
 
-  /**
-   * Method description
-   *
-   *
-   * @return
-   *
-   * @throws IOException
-   */
   List<Branch> getBranches() throws IOException;
+
+  default List<Branch> getBranchesWithStaleFlags(BranchStaleComputer computer) throws IOException {
+    Function<Branch, BranchStaleComputer.StaleContext> createContext = branch -> {
+      BranchStaleComputer.StaleContext staleContext = new BranchStaleComputer.StaleContext();
+      staleContext.setDefaultBranch(branch);
+      return staleContext;
+    };
+    List<Branch> branches = getBranches();
+    branches.stream()
+      .filter(Branch::isDefaultBranch)
+      .findFirst()
+      .ifPresent(defaultBranch -> branches.forEach(branch -> branch.setStale(computer.computeStale(branch, createContext.apply(defaultBranch)))));
+    return branches;
```

Comment:
In general this is fine but i find it kinda hard to read. I would have filtered for the default branch, initialized the StaleContext with the default branch and afterwards compute the branches within a foreach loop.

But you can leave it that way.